src/tests/unit/sync/sync_image_ring_abort_on_stopped_image.f90

```diff
@@ -11,15 +13,21 @@ program sync_images_ring
   if (num_images() .lt. 2) error stop "Need at least two images to test."
 
   associate (me => this_image())
-    if (me /= 1) then
-	  associate (lhs => merge(me - 1, num_images(), me /= 1), &
-				 rhs => merge(me + 1, 1, me /= num_images()))
-		sync images([lhs, rhs], STAT=stat_var)
-		! Only on image 2 and num_images() testing whether a stopped image is
-		! present can be done reliably. All other images could be up ahead.
-		if (stat_var /= STAT_STOPPED_IMAGE .and. me == 2) error stop "Error: stat_var /= STAT_STOPPED_IMAGE: "
-		if(me == 2) print *, 'Test passed.'
-	  end associate
-	end if
+    if (me == 1) then
+       continue !! image 1 does not participate and exits, creating a stopped image
+    else
+       associate (lhs => merge(me - 1, num_images(), me /= 1), &
+              rhs => merge(me + 1, 1, me /= num_images()))
+         sync images([lhs, rhs], STAT=stat_var)
+         !! Only images bordering image 1 (i.e., 2 and `num_images()`) can
+         !! accurately test whether a stopped image is present. All other
+         !! images could be up ahead.
+         if (stat_var /= STAT_STOPPED_IMAGE .and. me == 2) &
+              error stop "Error: stat_var /= STAT_STOPPED_IMAGE: "
+         if (stat_var /= STAT_STOPPED_IMAGE .and. me == num_images()) &
+              error stop "Error: stat_var /= STAT_STOPPED_IMAGE: "
```

Comment:
@vehre shouldn't this be OK too? It seems that this is failing on one Travis-CI OS X test (when code coverage flags are enabled): https://travis-ci.org/sourceryinstitute/opencoarrays/jobs/191684163#L463 Could there be an off-by-one error somewhere that I introduced, or in another part of the library/test?